blocklib/picoscope/Picoscope.hpp

```diff
@@ -231,20 +240,22 @@ class Picoscope : public PicoscopeBlockingHelper<TPSImpl, acquisitionMode == Acq
     A<float, "sample rate", Visible> sample_rate = 10000.f;
     // TODO any way to get custom enums into pmtv??
     A<std::string, "acquisition mode", Visible>                       acquisition_mode         = std::string("Streaming");
-    A<std::size_t, "pre-samples">                                     pre_samples              = 1000;
-    A<std::size_t, "post-samples">                                    post_samples             = 9000;
-    A<std::size_t, "no. captures (rapid block mode)">                 rapid_block_nr_captures  = 1;
+    A<gr::Size_t, "pre-samples">                                      pre_samples              = 1000;
+    A<gr::Size_t, "post-samples">                                     post_samples             = 9000;
+    A<gr::Size_t, "no. captures (rapid block mode)">                  rapid_block_nr_captures  = 1;
     A<bool, "trigger once (rapid block mode)">                        trigger_once             = false;
-    A<double, "poll rate (streaming mode)">                           streaming_mode_poll_rate = 0.001;
+    A<float, "poll rate (streaming mode)">                            streaming_mode_poll_rate = 0.001;
     A<bool, "auto-arm">                                               auto_arm                 = true;
     A<std::vector<std::string>, "IDs of enabled channels">            channel_ids;
     A<std::vector<std::string>, "Names of enabled channels">          channel_names;
     A<std::vector<std::string>, "Units of enabled channels">          channel_units;
-    std::vector<double>                                               channel_ranges;
-    std::vector<double>                                               channel_offsets;
+    A<std::vector<std::string>, "Quantity of enabled channels">       channel_quantities;
+    std::vector<float>                                                channel_ranges;
+    std::vector<float>                                                channel_offsets;
+    std::vector<float>                                                channel_gains; // only for floats
```

Comment:
Where are the 'channel_names', 'channel_units', 'channel_quantities' etc converted into tags? These would determine the automatic setup of sink names, axes, etc.

The `channel_ranges`, `channel_offsets`, `channel_gains` should have some annotations for user-facing API documentation.